src/buffer.c

```diff
@@ -501,51 +557,56 @@ buffer_set_initx(struct buffer_set_args *bsa)
         pthread_mutex_init(&bs->bs_mutex, NULL);
     }
 
-    int error = 0;
+    SIMPLE_LOG_MSG(LL_DEBUG, "s_region %p s_region_end 0x%lx size %ld "
+                   "bf_sz %ld bf_cnt %ld alignment %u",
+                   s_region, (uintptr_t)((char *)s_region + s_region_size),
+                   s_region_size, buf_size, nbufs, align);
+
+    size_t off = 0;
+    uintptr_t prev_end = (uintptr_t)NULL;
 
     for (size_t i = 0; i < nbufs; i++)
     {
         struct buffer_item *bi = calloc(1, sizeof(struct buffer_item));
         if (!bi)
         {
-            error = -ENOMEM;
-            break;
+            rc = -ENOMEM;
+            err_loc = 7;
+            goto xerror;
         }
 
         bi->bi_bs = bs;
         bi->bi_iov.iov_len = buf_size;
         bi->bi_register_idx = -1;
```

Comment:
Should we store the register_idx in the struct buffer_set now?  We could set it a special value like -1UL which would mean to get the idx value from the set.  Perhaps that's another PR?